app/src/main/java/org/septa/android/app/transitview/TransitViewResultsActivity.java

```diff
@@ -0,0 +1,574 @@
+        }
+        return true;
+    }
+
+    @Override
+    public void run() {
+        refreshed = true;
+        refreshData();
+    }
+
+    @Override
+    public void selectFirstRoute(RouteDirectionModel route) {
+        Log.e(TAG, "Invalid attempt to select the first route from the TransitViewResultsActivity -- going back to TransitView route picker");
+
+        Toast.makeText(this, R.string.transitview_add_route, Toast.LENGTH_LONG).show();
+        finish();
+    }
+
+    @Override
+    public void selectSecondRoute(RouteDirectionModel route) {
+        // sort the routes and append a null route to the end
+        List<RouteDirectionModel> selectedRoutes = new ArrayList<>();
+        selectedRoutes.add(firstRoute);
+        selectedRoutes.add(route);
+        Collections.sort(selectedRoutes, new RouteModelComparator());
+        selectedRoutes.add(null);
+
+        updateRouteLabels(selectedRoutes.get(0), selectedRoutes.get(1), selectedRoutes.get(2));
+    }
+
+    @Override
+    public void selectThirdRoute(RouteDirectionModel route) {
+        // sort the routes
+        List<RouteDirectionModel> selectedRoutes = new ArrayList<>();
+        selectedRoutes.add(firstRoute);
+        selectedRoutes.add(secondRoute);
+        selectedRoutes.add(route);
+        Collections.sort(selectedRoutes, new RouteModelComparator());
+
+        updateRouteLabels(selectedRoutes.get(0), selectedRoutes.get(1), selectedRoutes.get(2));
+    }
+
+    @Override
+    public void onMapReady(final GoogleMap googleMap) {
+        this.googleMap = googleMap;
+        MapStyleOptions mapStyle = MapStyleOptions.loadRawResourceStyle(this, R.raw.maps_json_styling);
+        googleMap.setMapStyle(mapStyle);
+
+        // hide navigation options and disable rotation
+        googleMap.getUiSettings().setMapToolbarEnabled(false);
+        googleMap.getUiSettings().setRotateGesturesEnabled(false);
+
+        // move camera to city hall to speed up zoom process
+        final LatLng cityHall = new LatLng(39.9517999, -75.1633285);
+        googleMap.moveCamera(CameraUpdateFactory.newLatLng(cityHall));
+
+        // default map zoom to show KML of all routes using builder.include()
+        LatLngBounds.Builder builder = new LatLngBounds.Builder();
+
+        // add to map of vehicle markers
+        details.clear();
+        for (String routeId : routeIds.split(",")) {
+            for (Map.Entry<TransitViewModelResponse.TransitViewRecord, LatLng> entry : parser.getResultsForRoute(routeId).entrySet()) {
+                String vehicleMarkerKey = new StringBuilder(routeId).append(VEHICLE_MARKER_KEY_DELIM).append(entry.getKey().getVehicleId()).toString();
+                details.put(vehicleMarkerKey, entry.getKey());
+
+                // map must include this vehicle when automatically moving
+                builder.include(entry.getValue());
+            }
+        }
+
+        final LatLngBounds bounds = builder.build();
```

Comment:
You don't seem to have any coordinates added to this bounds.

Reply:
I call builder.include() in line 221 to include all vehicles on the active route